### ISSUE TITLE
Lack of [[PlatformBrand]] is biting us

### ISSUE THREAD

**Issue by annevk** [whatwg/dom#597]
Consider https://dom.spec.whatwg.org/#concept-node-ensure-pre-insertion-validity. Step 4 bans `DocumentFragment`. Does it ban `ShadowRoot`? If it does, that's fine, but also a problem, since it shouldn't ban `CDATASection`.

I thought we had an issue tracking this, but it seems like we don't.

This is somewhat related to #457, which is about updating various algorithms to account for `Attr` nodes, which can now be passed to them again (and not all are anticipating that).

**Comment by annevk** [comment]
([[PlatformBrand]] is https://github.com/heycam/webidl/issues/97 by the way, which would give us a way more accurate way of describing what "_instance_ is an _interface_" actually means.)

**Comment by dwimmer90** [comment]
Hi,

Do you know how I can be removed from this group?

Danielle Wimmer

**Comment by annevk** [comment]
Use "Unwatch" in the GitHub UI (cannot be done via email).

**Comment by annevk** [comment]
I guess just like "exclusive Text node" we should have "exclusive DocumentFragment node" for now and use that. And add some tests.

**Comment by annevk** [comment]
Maybe this is not a problem. It seems Chrome and Safari happily treat a ShadowRoot as a DocumentFragment and don't really branch on it when appending. Probably still worth going through all the places, but this might be all mostly fine. (Would be good to check test coverage though.)
```
<!DOCTYPE html>
...<script>
x = document.createElement("div");
s = x.attachShadow({mode: "closed"});
is = s.appendChild(document.createElement("blah"));
document.body.appendChild(s);
</script>
```

**Comment by annevk** [comment]
I started looking into this to see if I could solve https://github.com/heycam/webidl/issues/659 (and also https://github.com/whatwg/dom/issues/636 and https://github.com/whatwg/dom/issues/719) and unfortunately it's not super straightforward.

1. I noticed that many instances of the term "node" reference the wrong definition. I have a patch for this (lacking `for=/`).
2. I noticed that we talk about Attr nodes, despite Attr not being defined as a node. 
3. If we go to the language of implements X, it doesn't really make sense anymore to distinguish between "abstract" interfaces and "concrete" ones. Or does it?

I'll upload a PR that takes a stab at this for further discussion.
